source/server/options_impl.cc

```diff
@@ -18,9 +18,24 @@
 #include "tclap/CmdLine.h"
 
 namespace Envoy {
+namespace {
+std::vector<std::string> toArgsVector(int argc, const char* const* argv) {
+  std::vector<std::string> args;
+  for (int i = 0; i < argc; ++i) {
+    args.emplace_back(argv[i]);
+  }
+  return args;
+}
+} // namespace
+
 OptionsImpl::OptionsImpl(int argc, const char* const* argv,
                          const HotRestartVersionCb& hot_restart_version_cb,
                          spdlog::level::level_enum default_log_level)
+    : OptionsImpl(toArgsVector(argc, argv), hot_restart_version_cb, default_log_level) {}
```

Comment:
I think that existing tests as it stands now are sufficient. However I would sleep much better if there was at least one unit test in the options_impl_test.cc that called the new constructor directly.

Reply:
I swapped the existing unit tests to use the new constructor, since the existing path of split-string-to-array-then-copy-to-vector was inefficient, and added some new tests of the argc/argv behavior.